src/repository/persistent.rs

```diff
@@ -209,16 +237,22 @@ impl<'manager> Repository<'manager> for PersistentRepository<'manager> {
             None => HashSet::new(),
         };
         heads.remove(&change_hash);
-        'outer: for change_hash in change.replace {
-            // TODO: store the dependant changes for each change to make this faster (for
-            // @CoCoSol007)
-            for entry in tx.prefix(&self.manager.changes, self.id.0.as_bytes()) {
-                let other: Change = borsh::from_slice(&entry?.1)?;
-                if other.replace.contains(&change_hash) {
-                    continue 'outer;
-                }
+        for replaced_hash in change.replace {
+            // Verify that the replaced change is replaced ONLY by this change.
+            let Ok(Some(replaced_change_by)) = tx.get(
+                &self.manager.dependants_changes,
+                borsh::to_vec(&(self.id, replaced_hash))?,
+            ) else {
+                return Err(Error::Io(io::Error::new(
+                    io::ErrorKind::NotFound,
```

Comment:
If the removed change replace a non existing change it should not return an error, but just remove the change.